connect/api/src/main/java/org/apache/kafka/connect/connector/Connector.java

```diff
@@ -121,4 +127,25 @@ public void reconfigure(Map<String, String> props) {
      * Stop this connector.
      */
     public abstract void stop();
+
+    /**
+     * Validate the connector configuration values against configuration definitions.
+     * @param connectorConfigs the provided configuration values
+     * @return List of Config, each Config contains the updated configuration information given
+     * the current configuration values.
+     */
+    public Config validate(Map<String, String> connectorConfigs) {
+        List<ConfigValue> configValues =  configDef.validate(connectorConfigs);
+        return new Config(configDef, groups, configValues);
+    }
+
+    /**
+     * Define the configuration for the connector.
+     * @return The ConfigDef for this connector.
+     */
+    protected abstract ConfigDef defineConfig();
+
+    protected List<String> defineGroup() {
```

Comment:
I'm going to flip flop on the group ordering point a bit. By putting it in the connector we're putting yet more burden on the connector developer. An implicit ordering based on the `ConfigDef.define()` calls could make sense too. What do you think? Is explicitly stating the order but requiring more implementation better, or should we try to figure this out implicitly and just rely on the more important groups to be created in the `ConfigDef` first?